test/functional/plugin/man_spec.lua

```diff
@@ -0,0 +1,181 @@
+local helpers = require('test.functional.helpers')(after_each)
+local plugin_helpers = require('test.functional.plugin.helpers')
+
+local Screen = require('test.functional.ui.screen')
+
+local buffer, command, eval = helpers.buffer, helpers.command, helpers.eval
+
+before_each(function()
+  plugin_helpers.reset()
+  helpers.clear()
+  command('syntax on')
+  command('set filetype=man')
+end)
+
+describe('In autoload/man.vim', function()
+  describe('function man#highlight_formatted_text', function()
+    local screen
+
+    before_each(function()
+      command('syntax off') -- Ignore syntax groups
+      screen = Screen.new(52, 5)
+      screen:attach()
+    end)
+
+    after_each(function()
+      screen:detach()
+    end)
+
+    local function expect(string)
+      screen:expect(string,
+      {
+        b = { bold = true },
+        i = { italic = true },
+        u = { underline = true },
+        bi = { bold = true, italic = true },
+        biu = { bold = true, italic = true, underline = true },
+      },
+      {{ bold = true, foreground = Screen.colors.Blue }})
+    end
+
+    local function expect_without_highlights(string)
```

Comment:
This is an anti-pattern. Just use `screen:snapshot_util()` and it will capture highlighting with not much extra work